### ISSUE TITLE
UI: show full resource type term in "resource type badge" instead of truncated and make it clickable

### ISSUE THREAD

**Issue by apirogov** [inveniosoftware/invenio-app-rdm#1200]
I was confused by a record that was shown as type "other", but did not show up when I searched and filtered for that resource type.

After inspecting the JSON, I figured out that this record is "publication-other", but the badge is the same.

I think that to avoid confusion it would be better that the badges would show the full resource type name, even though its more verbose. 

Furthermore, I think the badge should be clickable and forward to a search with that type pre-selected as a filter.

**Comment by lnielsen** [comment]
I agree about making the resource types badges linkable. 

Regarding naming, then it is [overridable](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/fixtures/data/vocabularies/resource_types.yaml#L320). The goal is that "other" is used as little as possible, so I'd rather that it says e.g. "Preprint" than "Preprint publication".

**Comment by lhenze** [comment]
I agree it would be useful and as-expected to make resource types badges clickable. 
What about "Access status"?  That could also be a link since it's also a filter.
